<a href="https://colab.research.google.com/github/Gerilda/Arrhythmia/blob/main/Arrhythmia.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [11]:
import numpy as np              # Массивы (матрицы, векторы, линейная алгебра)
import matplotlib.pyplot as plt # Научная графика
%matplotlib inline 
    # Говорим jupyter'у, чтобы весь графический вывод был в браузере, а не в отдельном окне
import pandas as pd             # Таблицы и временные ряды (dataframe, series)
import seaborn as sns           # Еще больше красивой графики для визуализации данных
import sklearn                  # Алгоритмы машинного обучения

###Аритмия

Цель состоит в том, чтобы различить наличие и отсутствие сердечной аритмии и классифицировать ее в одну из 16 групп.
Класс 01 относится к "нормальным" ЭКГ, классы 02-15 относятся к различным классам аритмий, а класс 16 относится к остальным неклассифицированным.
Эта база данных содержит 279 атрибутов, 206 из которых количественные (числовые), а остальные - номинальные.

In [12]:
url = "https://raw.githubusercontent.com/Gerilda/Arrhythmia/main/arrhythmia.csv"

In [13]:
data_raw = pd.read_csv(url, delimiter=';', na_values=['?'])

Вначале посмотрим на размеры таблицы - количество строк (каждая строка соответствует одной квартире) и количество столбцов (каждый столбец соответствует конкретному признаку):

In [14]:
data_raw.shape

(452, 280)

Отсюда видно, что в датасете есть 280 признаков.
Метод data_raw.head() отображает первые строки (по умолчанию, 5 строк) таблицы.

In [15]:
data_raw.head()

age  sex  height  weight  qrs_duration  ...   LD   LE    LF    LG  diagnosis
0   75    0     190      80            91  ...  0.9  2.9  23.3  49.4          8
1   56    1     165      64            81  ...  0.2  2.1  20.4  38.8          6
2   54    0     172      95           138  ...  0.3  3.4  12.3  49.0         10
3   55    0     175      94           100  ...  0.4  2.6  34.6  61.6          1
4   75    0     190      80            88  ... -0.1  3.9  25.4  62.8          7

[5 rows x 280 columns]

Отображаем последние строки таблицы (по умолчанию 5):

In [16]:
data_raw.tail()

age  sex  height  weight  qrs_duration  ...   LD   LE    LF    LG  diagnosis
447   53    1     160      70            80  ...  0.7  0.6  -4.4  -0.5          1
448   37    0     190      85           100  ...  0.4  2.4  38.0  62.4         10
449   36    0     166      68           108  ...  1.5  1.0 -44.2 -33.2          2
450   32    1     155      55            93  ...  0.5  2.4  25.0  46.6          1
451   78    1     160      70            79  ...  0.5  1.6  21.3  32.8          1

[5 rows x 280 columns]

###Attribute Information:

1 Age: Age in years , linear
2 Sex: Sex (0 = male; 1 = female) , nominal
3 Height: Height in centimeters , linear
4 Weight: Weight in kilograms , linear
5 QRS duration: Average of QRS duration in msec., linear
6 P-R interval: Average duration between onset of P and Q waves in msec., linear
7 Q-T interval: Average duration between onset of Q and offset of T waves in msec., linear
8 T interval: Average duration of T wave in msec., linear
9 P interval: Average duration of P wave in msec., linear
Vector angles in degrees on front plane of:, linear
10 QRS
11 T
12 P
13 QRST
14 J

15 Heart rate: Number of heart beats per minute ,linear

Of channel DI:
Average width, in msec., of: linear
16 Q wave
17 R wave
18 S wave
19 R' wave, small peak just after R
20 S' wave

21 Number of intrinsic deflections, linear

22 Existence of ragged R wave, nominal
23 Existence of diphasic derivation of R wave, nominal
24 Existence of ragged P wave, nominal
25 Existence of diphasic derivation of P wave, nominal
26 Existence of ragged T wave, nominal
27 Existence of diphasic derivation of T wave, nominal

Of channel DII:
28 .. 39 (similar to 16 .. 27 of channel DI)
Of channels DIII:
40 .. 51
Of channel AVR:
52 .. 63
Of channel AVL:
64 .. 75
Of channel AVF:
76 .. 87
Of channel V1:
88 .. 99
Of channel V2:
100 .. 111
Of channel V3:
112 .. 123
Of channel V4:
124 .. 135
Of channel V5:
136 .. 147
Of channel V6:
148 .. 159

Of channel DI:
Amplitude , * 0.1 milivolt, of
160 JJ wave, linear
161 Q wave, linear
162 R wave, linear
163 S wave, linear
164 R' wave, linear
165 S' wave, linear
166 P wave, linear
167 T wave, linear

168 QRSA , Sum of areas of all segments divided by 10, ( Area= width * height / 2 ), linear
169 QRSTA = QRSA + 0.5 * width of T wave * 0.1 * height of T wave. (If T is diphasic then the bigger segment is considered), linear

Of channel DII:
170 .. 179
Of channel DIII:
180 .. 189
Of channel AVR:
190 .. 199
Of channel AVL:
200 .. 209
Of channel AVF:
210 .. 219
Of channel V1:
220 .. 229
Of channel V2:
230 .. 239
Of channel V3:
240 .. 249
Of channel V4:
250 .. 259
Of channel V5:
260 .. 269
Of channel V6:
270 .. 279

Требуется предсказать категориальный признак `diagnosis`, представленный в числовом виде с номером типа по остальным признакам. Это *задача классификации*.

In [ ]:
plt.scatter(data_raw['Area'], data_raw['Price'])